app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -62,7 +62,10 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
         val safeIntent = intent.toSafeIntent()
         webRenderComponents.notifyLaunchWithSafeIntent(safeIntent)
 
-        lifecycle.addObserver(serviceLocator.engineViewCache)
+        arrayOf(
+            serviceLocator.engineViewCache,
+            serviceLocator.pocketVideoFetchScheduler
+        ).forEach { lifecycle.addObserver(it) }
```

Comment:
Nit: a little repetition here would actually be less code, and more immediately understandable.
```
lifecycle.addObserver(serviceLocator.engineViewCache)
lifecycle.addObserver(serviceLocator.pocketVideoFetchScheduler)
```